### PR TITLE
Feature/review fixes

### PR DESCRIPTION
**Pull request by MichaelKohler**

### CODE REVIEW

**Comment by MichaelKohler** [comment]
@freaktechnik r? if you want? ;)

File: shared/db/collections/sentences-meta.js [hunk 1]
```diff
@@ -144,7 +144,20 @@ export default class SentencesMeta {
...
+    // This works as all modern browsers use a stable sorting algorithm
+    // if not, we're still fine..
+    // We always want to show sentences with more valid votes first
+    const additionallySortedByApprovalVotes = sentences.sort((a, b) => {
+      const aVoteLength = a.valid.length;
+      const bVoteLength = b.valid.length;
+
+      if (aVoteLength < bVoteLength) return 1;
+      if (aVoteLength > bVoteLength) return -1;
+      return 0;
```
Comment (freaktechnik) [inline shared/db/collections/sentences-meta.js]: `return bVoteLength - aVoteLength;`

File: web/src/components/review-form.jsx [hunk 2]
```diff
@@ -120,7 +128,13 @@ export default class ReviewForm extends React.Component {
...
         )) }
 
-        <button type="submit">Finish Review</button>
+        { this.state.pendingSentences && (
+          <p>
+            <strong>Reviews are being uploaded. This can take several minutes depending on the number of sentences added.
+            Please don't close this website.</strong>
+          </p>
+        )}
+        <button type="submit" disabled={this.state.pendingSentences}>Finish Review</button>
```
Comment (freaktechnik) [inline web/src/components/review-form.jsx]: Does this also disable submit by keyboard of the form?

File: shared/db/collections/sentences-meta.js (hunk 1 above)
Comment (freaktechnik) [reply shared/db/collections/sentences-meta.js]: Or of you only want 1 - -1 use Math.sgn to shorten.

File: web/src/components/review-form.jsx (hunk 2 above)
Comment (MichaelKohler) [reply web/src/components/review-form.jsx]: while not being 100% sure, I'd say this should be prevented by the browser..

**Pull request merged**